main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+    _ "fmt"
+    "os"
+    "io/ioutil"
+    "bytes"
+    "flag"
+    "bufio"
+)
+
+
+func main() {
+    /* Go won't allow the same arg/flag construction we had before, so all flags instead */
+    project := flag.String("project", "", "Project name")
+    multisite := flag.Bool("multisite", false, "Make it a multisite")
+    flag.Parse()
+    if *project == "" {
+        println("You must set a value for the project argument")
+        os.Exit(1)
+    }
+    if *multisite == true {
+        println("MULTISITE")
+    }
+
+    makedir("bin")
+    makedir("setup")
+    makedir("setup/content")
+
+    createDockerCompose(*project)
+    createBinWp()
+    createBinConsole()
+    createBinSetup()
+    createExternal()
+    createInternal(*multisite)
+}
+
+func createDockerCompose(project string) {
+    dockerComposeContents := dockerComposeContents()
+    dockerComposeContents = findAndReplace(dockerComposeContents, []byte("!!!PROJECTNAME!!!"), []byte(project))
+    creatingNoexec("docker-compose.yml", dockerComposeContents)
+}
+
+func createBinWp() {
+    contents := wpContents()
+    creating("bin/wp", contents)
+}
+
+func createBinConsole() {
+    contents := consoleContents()
+    creating("bin/console", contents)
+}
+
+func createBinSetup() {
+    contents := setupContents()
+    creating("bin/setup", contents)
+}
+
+func createExternal() {
+    contents := externalContents()
+    creating("setup/external.sh", contents)
+}
+
+func createInternal(multisite bool) {
+    contents := internalContents()
+    if multisite {
+        contents = findAndReplace(contents, []byte("!!!INSTALLTYPE!!!"), []byte("multisite-install"))
+        contents = findAndReplace(contents, []byte("!!!ACTIVATIONTYPE!!!"), []byte("--network"))
+        contents = findAndReplace(contents, []byte("!!!THEMEENABLE!!!"), []byte("wp theme enable --network $theme"))
+    } else {
+        contents = findAndReplace(contents, []byte("!!!INSTALLTYPE!!!"), []byte("install"))
+        contents = findAndReplace(contents, []byte("!!!ACTIVATIONTYPE!!!"), []byte(""))
+        contents = findAndReplace(contents, []byte("!!!THEMEENABLE!!!"), []byte(""))
+    }
+    creating("setup/internal.sh", contents)
+}
+
+func makedir(directory string) {
+    println("-> Creating", directory)
+    os.Mkdir(directory, 0755)
```

Comment:
If we call `makedir()` from `creating()` then it would make sense to use `os.MkdirAll()` here. So we don't need to worry about the order things are created in.